src/parser/warlock/demonology/modules/pets/DemoPets/PetSummonHandler.js

```diff
@@ -47,6 +48,13 @@ class PetSummonHandler extends Analyzer {
       // Wild Imps need few additional properties
       pet.setWildImpProperties(this._lastPlayerPosition);
     }
+    if (petInfo.name === "Demonic Tyrant") {
```

Comment:
Actually this should have a Demonic Consumption talent check here too. The meta shouldn't get set when you don't have DT => you don't consume imps => you don't get empowered.